src/libmongoc/src/mongoc/mongoc-cursor.c

```diff
@@ -1449,9 +1469,16 @@ void
 mongoc_cursor_set_batch_size (mongoc_cursor_t *cursor, uint32_t batch_size)
 {
    BSON_ASSERT (cursor);
+   bson_iter_t iter;
 
-   _mongoc_cursor_set_opt_int64 (
-      cursor, MONGOC_CURSOR_BATCH_SIZE, (int64_t) batch_size);
+   if (bson_iter_init_find (&iter, &cursor->opts, MONGOC_CURSOR_BATCH_SIZE) &&
+       BSON_ITER_HOLDS_INT32 (&iter)) {
+      BSON_ASSERT (bson_in_range_int32_t_unsigned (batch_size));
```

Comment:
@kevinAlbs Are we sure about this assertion? This is effectively adding a new _precondition_ to this function that the input value is conditionally representable by whatever type `batchSize` happens to be set with. I think it may be more appropriate for this to be a `MONGOC_WARNING` instead.

Reply:
Good catch. I agree using `MONGOC_WARNING` seems more appropriate. I think a caller could expect `mongoc_cursor_set_batch_size (UINT32_MAX)` to succeed if the stored `batchSize` is represented with an int64.

Suggest using a warning and returning early if not in range:
```c
      if (!bson_in_range_int32_t_unsigned (batch_size)) {
         MONGOC_WARNING ("unable to overwrite stored int32 batchSize with "
                         "out-of-range value %" PRIu32,
                         batch_size);
         return;
      }
```

Reply:
Bump: replace assertion with `MONGOC_WARNING` to avoid narrowing the contract of this function.